### PR TITLE
Remove pull_request from actions

### PR DESCRIPTION
**Pull request by jcreedcmu**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: Build Extension
-on: [push, pull_request]
+on: push
```
Comment (aeisenberg) [inline .github/workflows/main.yml:2]: Are we sure we want to do this? And if we do, we need this:

```
  on: [push]
```

or 

```
  on:
    push:
```

**Comment by jcreedcmu** [comment]
The motivation for doing it was @adityasharad saying pull_request wasn't supported. You're correct about my yaml error, though, will fix.

**Comment by alexet** [comment]
It's only not supported for the codeql action. It is fine for the checks action and ensures that the tests are run on PRs from users who haven't enabled actions on their fork.

**Comment by jcreedcmu** [comment]
Ah, ok, so I misunderstood that, then. Closing as this is no longer necessary.

**Comment by alexet** [comment]
You may still want to delete it from `codeql.yml`

**Comment by jcreedcmu** [comment]
Correct; but https://github.com/github/vscode-codeql/pull/405 is already doing that.

**Pull request closed without merging**